Userland: Tighten API to DeprecatedFile to prevent accidental new usages

BenWiederhake opened the pull request:
This PR:
- Merges `DEFAULT_PATH` to a new home in `Core::File`, instead of having it in two different places
- Removes two unused functions from `DeprecatedFile`, to ensure we don't accidentally use more of this old API.

Advances #17129.

timschumi commented on Userland/Libraries/LibCore/File.h:
```diff
@@ -17,6 +17,9 @@
 
 namespace Core {
 
+#define DEFAULT_PATH "/usr/local/sbin:/usr/local/bin:/usr/bin:/bin"
+#define DEFAULT_PATH_SV "/usr/local/sbin:/usr/local/bin:/usr/bin:/bin"sv
```
This feels more like a `Core::FileSystem` thing, I just put it in `Core::DeprecatedFile` originally because we didn't have a better alternative back then (and therefore it was subsequently copied to `FileSystem`).

BenWiederhake replied:
*shrug* Then I'll merge it into FileSystem

BenWiederhake replied:
Ah! No, sorry, there is a reason against this: Sometimes this introduces a dependency on LibFileSystem that wouldn't exist before, so I'd like to avoid that.

BenWiederhake replied:
Hmm. the only example for "reason against this" is `LibC/unistd.cpp`, which already has the line `#include <LibFileSystem/FileSystem.h>`, so that argument is invalid.

Yup, I'll move it to LibFileSystem.

BenWiederhake commented:
Changes since last push:
- Moved `DEFAULT_PATH` to `LibFileSystem/FileSystem.h` instead of `Core/File.h`, thanks to @timschumi!

BenWiederhake commented:
Changes since last push:
- Rebased
- (Nothing else. Basically I just wanted to re-run CI, because for some reason the discord notification failed, which caused this PR to be red.)

awesomekling commented:
The commit message "Userland: Merge DEFAULT_PATH into Core/File.h" seems at odds with what is actually being changed :)

konradekk reviewed (approved):
removing some code canʼt be too bad! 😅
